tests/src/Test/Tasty/Vivado.hs

```diff
@@ -0,0 +1,53 @@
+{-# LANGUAGE TypeApplications #-}
+
+module Test.Tasty.Vivado where
+
+import Data.Coerce (coerce)
+import Data.Tagged (Tagged (..))
+import Data.Proxy (Proxy (..))
+import System.IO (hPutStr, hFlush)
+
+import System.IO.Temp (withSystemTempFile)
+import Test.Tasty.Providers (IsTest (..), testPassed)
+import Test.Tasty.Options (IsOption (..), safeReadBool, flagCLParser, lookupOption, OptionDescription (..))
+
+import Clash.Vivado (fromModuleName, HdlSource (..))
+import Clash.Annotations.Primitive (HDL (..))
+import Test.Tasty.Program
+
+-- | @--vivado@ flag for enabling tests that use vivado.
+newtype Vivado = Vivado Bool
+
+instance IsOption Vivado where
+  defaultValue = Vivado True
+  parseValue = fmap Vivado . safeReadBool
+  optionName = pure "no-vivado"
+  optionHelp = pure "Skip Vivado tests"
+  optionCLParser = flagCLParser Nothing (Vivado False)
+
+data VivadoTest = VivadoTest
+  { hdlTarget :: HDL
+  , hdlSource :: IO FilePath
+  , moduleTest :: String
+  , entity :: String
+  }
+
+instance IsTest VivadoTest where
+  run optionSet (VivadoTest target dir m top) progressCallback
+    | Vivado True <- lookupOption optionSet =
+        withSystemTempFile "sim.tcl" $ \fp h -> do
+          hdlSrc <- dir
+          tcl <- fromModuleName (HdlSource hdlSrc target) m top
+          hPutStr h tcl *>
+            hFlush h
+          runVivado hdlSrc ["-mode", "batch", "-source", fp]
+    | otherwise = pure (testPassed "Ignoring test due to --no-vivado")
+
+   where
+    vivado workDir args =
+      TestFailingProgram True "vivado" args NoGlob PrintNeither False (Just 0) (ExpectNotStdOut "expected: ") (Just workDir)
```

Comment:
```suggestion
      TestFailingProgram True "vivado" args NoGlob PrintNeither False (Just 1) (ExpectNotStdOut "expected: ") (Just workDir)
```

I think I messed this up. This used to be `1` in your code @vmchale, but I changed it to `0` for debugging purposes and mindlessly squashed..

Reply:
Oh no! I think this is supposed to be 0, the `1` was in there for debugging purposes. `0` is exit success!

Reply:
agreed!